lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -341,13 +344,14 @@ class TwilioConnectionTransport extends StateMachine {
 
   /**
    * Publish (or queue) an event to the Insights gateway.
-   * @param {string} groupName - Event group name
-   * @param {string} eventName - Event name
+   * @param {string} group - Event group name
+   * @param {string} name - Event name
+   * @param {string} level - Event level
    * @param {object} payload - Event payload
-   * @returns {boolean} true if queued or published, false if disconnected from the Insights gateway
+   * @returns {void}
    */
-  publishEvent(groupName, eventName, payload) {
-    return this._eventPublisher.publish(groupName, eventName, payload);
+  publishEvent(group, name, level, payload) {
+    this._eventObserver.emit('event', { group, name, level, payload });
```

Comment:
I do not like this `emit` pattern - where consumers of eventObserver are emitting on it. but I understand that its not exposed by this PR. We should clean this up when we get rid of `eventObserver` conenct option..

Reply:
@makarandp0 In this case, what we are doing is legitimate because the EventObserver is supposed to observe events emitted on it by the consumer. This is different from EventEmitter, where the consumer observes events emitted by it.